_source/_posts/2021-09-24-java-records.md

```diff
@@ -0,0 +1,556 @@
+    @Override
+    public LocalDate convert(String s) {
+        return LocalDate.parse(s);
+    }
+}
+```
+```java
+package com.okta.developer.records.repository;
+
+
+import org.springframework.core.convert.converter.Converter;
+
+import java.time.LocalTime;
+
+public class LocalTimeConverter implements Converter<String, LocalTime> {
+    @Override
+    public LocalTime convert(String s) {
+        return LocalTime.parse(s);
+    }
+}
+```
+
+Add the class `MongoConfiguration` in the package `com.okta.developer.records.configuration`, to register the converters:
+
+```java
+package com.okta.developer.records.configuration;
+
+import com.okta.developer.records.repository.LocalDateConverter;
+import com.okta.developer.records.repository.LocalTimeConverter;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.mongodb.config.AbstractReactiveMongoConfiguration;
+import org.springframework.data.mongodb.core.convert.MongoCustomConversions;
+
+@Configuration
+public class MongoConfiguration extends AbstractReactiveMongoConfiguration {
+
+    @Value("${spring.data.mongodb.database}")
+    private String databaseName;
+
+    @Override
+    protected String getDatabaseName() {
+        return databaseName;
+    }
+
+    @Override
+    protected void configureConverters(MongoCustomConversions.MongoConverterConfigurationAdapter adapter) {
+        adapter.registerConverter(new LocalDateConverter());
+        adapter.registerConverter(new LocalTimeConverter());
+    }
+}
+```
+Add the `StatsController` under the `com.okta.developer.records.controller` package:
+
+```java
+package com.okta.developer.records.controller;
+
+import com.okta.developer.records.domain.EndOfGame;
+import com.okta.developer.records.domain.MentalStateDamage;
+import com.okta.developer.records.repository.StatsRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+import reactor.core.publisher.Flux;
+
+@RestController
+public class StatsController {
+
+    @Autowired
+    private StatsRepository statsRepository;
+
+    @GetMapping("/endOfGame")
+    public Flux<EndOfGame> getAllEndOfGame(){
+        return statsRepository.findAll();
+    }
+
+    @GetMapping("/mentalStateAverageDamage")
+    public Flux<MentalStateDamage> getMentalStateAverageDamage(){
+        return statsRepository.queryMentalStateAverageDamage();
+    }
+
+}
+```
+
+The controller enables the `/endOfGame` endpoint to get all entries, and the `/mentalStateAverageDamage` endpoint, that returns the damage in each category, in an average by mental state.
+
+Download the test dataset from [Github](https://github.com/indiepopart/java-records/blob/main/src/test/resources/stats.json), and copy it to `src/test/resouces/stats.json`.
```

Comment:
If possible, please add an HTTPie command below to show how to do this. That way, it's easy to copy/paste when doing the tutorial.

Fun fact: I just learned HTTPie has an `https` command you can use so you don't have to specify `https://` for hosts.